src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -74,12 +84,163 @@ private void Initialize()
+                ImmutableDictionary.Create<ModuleDeclarationSyntax, DiagnosticBuilder.ErrorBuilderDelegate>());
+
+            Compilation compilation = new Compilation(AzResourceTypeProvider.CreateWithAzTypes(), syntaxTreeGrouping);
+            SemanticModel semanticModel = compilation.GetEntrypointSemanticModel();
+
+            return ResourceDependencyVisitor.GetResourceDependencies(semanticModel);
+        }
+
+        public Snippet GetResourceBodyCompletionSnippet(TypeSymbol typeSymbol)
+        {
+            string label = "{}";
+            string type = typeSymbol.Name;
+
+            StringBuilder sb = new StringBuilder();
+
+            // Get resource body completion snippet from checked in static template file, if available
+            if (resourceTypeToBodyMap.TryGetValue(type, out string? resourceBody))
+            {
+                sb.AppendLine(resourceBody);
+
+                if (resourceTypeToDependentsMap.TryGetValue(type, out string? resourceDependencies))
+                {
+                    sb.Append(resourceDependencies);
+                }
+
+                return new Snippet(sb.ToString(), CompletionPriority.Medium, label, label);
+            }
+
+            // Get resource body completion snippet from swagger spec
+            return GetResourceBodyCompletionSnippetFromSwaggerSpec(typeSymbol, label);
+        }
+
+        private Snippet GetResourceBodyCompletionSnippetFromSwaggerSpec(TypeSymbol typeSymbol, string label)
+        {
+            if (typeSymbol is ResourceType resourceType && resourceType.Body is ObjectType objectType)
+            {
+                int index = 1;
+                string? snippetText = null;
+
+                foreach (KeyValuePair<string, TypeProperty> kvp in objectType.Properties.OrderBy(x => x.Key))
+                {
+                    snippetText = GetSnippetText(kvp.Value, snippetText, ref index);
+                }
+
+                if (snippetText is not null)
+                {
+                    // Define final tab stop
+                    snippetText += "\t$0\n}";
+
+                    // Add to cache
+                    resourceTypeToBodyMap.Add(typeSymbol.Name, snippetText);
+
+                    return new Snippet(snippetText, CompletionPriority.Medium, label, label);
+                }
+            }
+
+            return new Snippet("{\n\t$0\n}", CompletionPriority.Medium, label, label);
+        }
+
+        private string? GetSnippetText(TypeProperty typeProperty, string? snippetText, ref int index)
+        {
+            if (typeProperty.Flags.HasFlag(TypePropertyFlags.Required))
+            {
+                if (snippetText is null)
+                {
+                    snippetText = "{\n";
+                }
+
+                if (typeProperty.TypeReference.Type is ObjectType objectType)
+                {
+                    snippetText += "\t" + typeProperty.Name + ": {\n";
```

Comment:
>"\t" [](http://example.com/codeflow?start=35&length=4)

I think you will need to keep track of the indent level to help VS code format the nested properties correctly via `AdjustIndentation`

Reply:
This is related to one of the screenshots I posted.

Reply:
Fixed formatting here:
https://github.com/Azure/bicep/commit/c1e6263ec2141e71df6b2cf96040955ff8700b1f